File: Bit manupulation/Property based problems/Count set bits/brute_force_app.cpp
#include<bits/stdc++.h>
using namespace std;

// Time complextiy :- O(N)
// Space complexity :- O(1)

// Approach :-   
// loop from 1 to n count the number of set bit in each number 

//Link :- https://www.geeksforgeeks.org/problems/count-total-set-bits-1587115620/1

// Function to get no of set bits in binary
// representation of positive integer n
int setBits(int n) {
	int count = 0;
	while (n) {
		count += n & 1;
		n >>= 1;
	}
	return count;
}

// Count total set bits in all 
// numbers from 1 to n
int countSetBits(int n)  {
   int res = 0;
   for (int i=1; i<=n; i++)
      res += setBits(i);
   return res;
}

File: Queue/Minimum K Consecutive Bit Flips/brute_force_app.java
import java.util.*

//Time complexity :- O(n*k)
//Space complexity :- O(1)

// Approach :- 
// check if the current number is 0, the flip the next k bit 
// flip until there are k bit on the right side of the current bit 
// and if rest all are 1 after flipping then it's possible, count the steps 


//Link :- https://www.geeksforgeeks.org/problems/minimum-number-of-k-consecutive-bit-flips--171650/1

class Solution {
    public int kBitFlips(int[] arr, int k) {
        int n = arr.length, res = 0;
        
        for(int i=0; i<n-k+1; ++i){
            if(arr[i] == 0){
                for(int j=i; j<(i+k); ++j){
                    arr[j] = 1^arr[j];
                }
                res += 1;
            }
        }
        
        for(int i=n-k+1; i<n; ++i){
            if(arr[i] == 0){
                return -1;
            }
        }
        return res;
    }
}